classes/classes/Scenes/Monsters/pregnancies/PlayerBunnyPregnancy.as

```diff
@@ -0,0 +1,208 @@
+package classes.Scenes.Monsters.pregnancies 
+{
+	import classes.GlobalFlags.kGAMECLASS;
+	import classes.Items.Mutations;
+	import classes.Player;
+	import classes.PregnancyStore;
+	import classes.Scenes.PregnancyProgression;
+	import classes.Scenes.VaginalPregnancy;
+	import classes.StatusEffects;
+	import classes.internals.GuiOutput;
+	import classes.internals.PregnancyUtils;
+	
+	/**
+	 * Contains pregnancy progression and birth scenes for a Player impregnated by a bunny.
+	 */
+	public class PlayerBunnyPregnancy implements VaginalPregnancy 
+	{
+		private var pregnancyProgression:PregnancyProgression;
+		private var output:GuiOutput;
+		private var mutations:Mutations;
+		
+		/**
+		 * Create a new bunny pregnancy for the player. Registers pregnancy for bunnies.
+		 * @param	pregnancyProgression instance used for registering pregnancy scenes
+		 * @param	output instance for gui output
+		 */
+		public function PlayerBunnyPregnancy(pregnancyProgression:PregnancyProgression, output:GuiOutput, mutations:Mutations)
+		{
+			this.pregnancyProgression = pregnancyProgression;
+			this.output = output;
+			this.mutations = mutations;
+			
+			pregnancyProgression.registerVaginalPregnancyScene(PregnancyStore.PREGNANCY_PLAYER, PregnancyStore.PREGNANCY_BUNNY, this);
+		}
```

Comment:
This should do as well:
```as3
		public function PlayerBunnyPregnancy(pregnancyProgression:PregnancyProgression)
		{
			this.pregnancyProgression = pregnancyProgression;
			output = Output.init();
			mutations = Mutations.init();
			
			pregnancyProgression.registerVaginalPregnancyScene(PregnancyStore.PREGNANCY_PLAYER, PregnancyStore.PREGNANCY_BUNNY, this);
		}
```

Well, just a suggestion, no 'must do'.

Reply:
Unfortunately, the example you provided `output = Output.init();` is a textbook case of Singleton abuse.

Singletons should be used when only one instance of a class should exist. Unfortunately they are often misused for global state and 'easy' access, as you just demonstrated in the example.
This leads to classes lying about their dependencies.

Related is the rampant inheritance abuse in the form of `BaseContent`.
The reason for instance why the call `new Creature();`
will fail without 
```actionscript
		[BeforeClass]
		public static function setUpClass():void {
			kGAMECLASS = new CoC(StageLocator.stage);
		}
```

Further reading:
- [Singletons are Pathological Liars, Miško Hevery](http://misko.hevery.com/2008/08/17/singletons-are-pathological-liars/)
- [When is Singleton appropriate?, Stack Exchange](https://softwareengineering.stackexchange.com/questions/252/when-is-singleton-appropriate)

Reply:
@brrritssocold wrote:
> Singletons should be used when only one instance of a class should exist.

Actually that was the intention of making those two into a singleton.